### PR TITLE
Write config UI

### PR DESCRIPTION
**Pull request by CannonLock**
- Adds ability to edit the config
- Type checking where possible

### CODE REVIEW

**Comment by CannonLock** [comment]
Closes #753 and #322

File: web_ui/frontend/app/config/main.d.ts
Comment (haoming29) [inline web_ui/frontend/app/config/main.d.ts:1]: This is file required as a placeholder?

File: docs/parameters.yaml
Comment (haoming29) [inline docs/parameters.yaml:1]: Not related to code changes in this PR but I figured that `Xrootd.ScitokensConfig` should be for `origin` and `cache` only.

**Review by haoming29 — APPROVED** [review]
Two minor changes requested and are non-blocking so feel free to tackle them or leave them for now but otherwise looks pretty nice to me!

I'm thinking it would be very nice if we could add "Cancel" button next to save, so that users mistakenly changing a value have the chance to ignore the change. Could be just a button to refresh the page if that's easier to implement.

As we talked in person, currently if a user changes a value and re-do the change, frontend will still send that change back to backend, which won't be a big issue now but we can fix them in another PR.

Screenshot 2024-03-07 at 12 56 23 PM

**Pull request merged**